src/crate/theme/vendor/rtd_compat/extension.py

```diff
@@ -0,0 +1,242 @@
+import os
+
+from sphinx import version_info
+from sphinx.util import logging
+
+import requests
+
+from .utils import get_github_username_repo, get_bitbucket_username_repo, get_gitlab_username_repo
+
+
+logger = logging.getLogger(__name__)
+
+
+# https://www.sphinx-doc.org/en/stable/extdev/appapi.html#event-html-page-context
+def manipulate_config(app, config):
+    logger.info(
+        'Running "manipulate_config" from Read the Docs "sphinx_build_compatibility" extension. '
+        'Consider removing it from your requirements and migrating your documentation accordingly. '
+        'This extension is useful only as a transition but it will not be maintained.'
+    )
+
+    # Add Read the Docs' static path.
+    # Add to the end because it overwrites previous files.
+    if not hasattr(config, "html_static_path"):
+        config.html_static_path = []
+    if os.path.exists('_static'):
+        config.html_static_path.append('_static')
+
+    # Define this variable in case it's not defined by the user.
+    # It defaults to `alabaster` which is the default from Sphinx.
+    # https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_theme
+    if not hasattr(config, "html_theme"):
+        config.html_theme = 'alabaster'
+
+    # Example: ``/docs/``
+    conf_py_path = "/"
+    conf_py_path += os.path.relpath(
+            str(app.srcdir),
+            os.environ.get("READTHEDOCS_REPOSITORY_PATH"),
+        ).strip("/")
+    conf_py_path += "/"
+
+    git_clone_url = os.environ.get("READTHEDOCS_GIT_CLONE_URL")
+    github_user, github_repo = get_github_username_repo(git_clone_url)
+    bitbucket_user, bitbucket_repo = get_bitbucket_username_repo(git_clone_url)
+    gitlab_user, gitlab_repo = get_gitlab_username_repo(git_clone_url)
+
+    project_slug = os.environ.get("READTHEDOCS_PROJECT")
+    version_slug = os.environ.get("READTHEDOCS_VERSION")
+    production_domain = os.environ.get("READTHEDOCS_PRODUCTION_DOMAIN", "readthedocs.org")
+
+    scheme = "https"
+    if production_domain.startswith("devthedocs"):
+        scheme = "http"
+
+    # We are using APIv2 to pull active versions, downloads and subprojects
+    # because APIv3 requires a token.
+    try:
+        response_versions = requests.get(
+            f"{scheme}://{production_domain}/api/v2/version/?project__slug={project_slug}&active=true",
+            timeout=2,
+        ).json()
+        versions = [
+            (version["slug"], f"/{version['project']['language']}/{version['slug']}/")
+            for version in response_versions["results"]
+        ]
```

Comment:
On the [preview for crate-reference](https://crate--16732.org.readthedocs.build/en/16732/), intending to restore the version chooser, it looks like this now:

![Image](https://github.com/user-attachments/assets/b44dd6ad-dd54-43ab-a4f1-4139450a8de6)

Now, the question is if it will be all good on the live site, or if it will also start displaying all the PR builds, for whatever reason, which we certainly want to avoid.

Reply:
Well, yeah. It would probably also happen on the live site.

```bash
http 'https://readthedocs.org/api/v2/version/?project__slug=crate&active=true' | jq '.results[].slug'
```

```json
"master"
"latest"
"5.8"
"5.7"
"4.8"
"3.3"
"16732"
"16731"
"16729"
"16728"
```

Reply:
Maybe [APIv2](https://docs.readthedocs.io/en/stable/api/v2.html#version-list) has a bug, but I couldn't figure out the right parameters.

[APIv3](https://docs.readthedocs.io/en/stable/api/v3.html#versions-listing) however works well.
```
$ http 'https://readthedocs.org/api/v3/projects/crate/versions/?active=true' | jq '.results[].slug'
"master"
"latest"
"5.8"
"5.7"
"4.8"
"3.3"
```

Reply:
c0c0929 adds a fix by adjusting the RTD compat layer. Now, this spot looks like before.

![image](https://github.com/user-attachments/assets/0d510834-cbc6-4d83-ab24-b28f7fc2e118)